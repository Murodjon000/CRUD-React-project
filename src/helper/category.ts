const categories = ['High', 'Medium', 'Low'];

export default categories;
