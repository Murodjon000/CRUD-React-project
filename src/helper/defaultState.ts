export const defaultState = [
    {
      id: '12252',
      title: 'Where can I get some?',
      description:
        "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
      image: 'https://miro.medium.com/max/1000/1*9xsPgsjcFOnwhgPwtsO_HQ.jpeg',
      category: 'High',
      editing: false,
    },
    {
      id: '12212352',
      title: 'Why do we use it?',
      description:
        "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
      image:
        'https://cdn.searchenginejournal.com/wp-content/uploads/2019/08/managing-global-search-projects-760x400.png',
      category: 'Medium',
      editing: false,
    },
    {
      id: '23',
      title: 'What is Lorem Ipsum?',
      description:
        "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
      image:
        'https://www.oneaccountingsoftware.com/au/wp-content/uploads/2019/08/How-to-Prepare-Annual-Performance-Report.jpg',
      category: 'Low',
      editing: false,
    },
  ];