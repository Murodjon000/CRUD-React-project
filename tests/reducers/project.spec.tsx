import projectReducer from "../../src/reducers/project";
import { defaultState } from "../../src/helper/defaultState";

describe("projects reducer", () => {
  it("should return the initial state", () => {
    //eslint-disable-next-line
    expect(
      projectReducer(undefined, { payload: {}, type: "" })
    ).toMatchSnapshot();
  });

  it("should handle CREATE_PROJECT", () => {
    expect(
      projectReducer(defaultState, {
        type: "CREATE_PROJECT",
        payload: { defaultState },
      })
    ).toMatchSnapshot();
  });

  it("should handle REMOVE_PROJECT", () => {
    expect(
      projectReducer(defaultState, {
        type: "REMOVE_PROJECT",
        payload: { defaultState },
      })
    ).toMatchSnapshot();
  });

  it("should handle EDIT_PROJECT", () => {
    expect(
      projectReducer(defaultState, {
        type: "EDIT_PROJECT",
        payload: { defaultState },
      })
    ).toMatchSnapshot();
  });

  it("should handle UPDATE_PROJECT", () => {
    expect(
      projectReducer(defaultState, {
        type: "UPDATE_PROJECT",
        payload: { defaultState },
      })
    ).toMatchSnapshot();
  });
});
