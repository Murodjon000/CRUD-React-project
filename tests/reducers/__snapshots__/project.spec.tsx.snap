// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`projects reducer should handle CREATE_PROJECT 1`] = `
Array [
  Object {
    "category": "High",
    "description": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
    "editing": false,
    "id": "12252",
    "image": "https://miro.medium.com/max/1000/1*9xsPgsjcFOnwhgPwtsO_HQ.jpeg",
    "title": "Where can I get some?",
  },
  Object {
    "category": "Medium",
    "description": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
    "editing": false,
    "id": "12212352",
    "image": "https://cdn.searchenginejournal.com/wp-content/uploads/2019/08/managing-global-search-projects-760x400.png",
    "title": "Why do we use it?",
  },
  Object {
    "category": "Low",
    "description": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
    "editing": false,
    "id": "23",
    "image": "https://www.oneaccountingsoftware.com/au/wp-content/uploads/2019/08/How-to-Prepare-Annual-Performance-Report.jpg",
    "title": "What is Lorem Ipsum?",
  },
  undefined,
]
`;

exports[`projects reducer should handle EDIT_PROJECT 1`] = `
Array [
  Object {
    "category": "High",
    "description": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
    "editing": false,
    "id": "12252",
    "image": "https://miro.medium.com/max/1000/1*9xsPgsjcFOnwhgPwtsO_HQ.jpeg",
    "title": "Where can I get some?",
  },
  Object {
    "category": "Medium",
    "description": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
    "editing": false,
    "id": "12212352",
    "image": "https://cdn.searchenginejournal.com/wp-content/uploads/2019/08/managing-global-search-projects-760x400.png",
    "title": "Why do we use it?",
  },
  Object {
    "category": "Low",
    "description": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
    "editing": false,
    "id": "23",
    "image": "https://www.oneaccountingsoftware.com/au/wp-content/uploads/2019/08/How-to-Prepare-Annual-Performance-Report.jpg",
    "title": "What is Lorem Ipsum?",
  },
]
`;

exports[`projects reducer should handle REMOVE_PROJECT 1`] = `
Array [
  Object {
    "category": "High",
    "description": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
    "editing": false,
    "id": "12252",
    "image": "https://miro.medium.com/max/1000/1*9xsPgsjcFOnwhgPwtsO_HQ.jpeg",
    "title": "Where can I get some?",
  },
  Object {
    "category": "Medium",
    "description": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
    "editing": false,
    "id": "12212352",
    "image": "https://cdn.searchenginejournal.com/wp-content/uploads/2019/08/managing-global-search-projects-760x400.png",
    "title": "Why do we use it?",
  },
  Object {
    "category": "Low",
    "description": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
    "editing": false,
    "id": "23",
    "image": "https://www.oneaccountingsoftware.com/au/wp-content/uploads/2019/08/How-to-Prepare-Annual-Performance-Report.jpg",
    "title": "What is Lorem Ipsum?",
  },
]
`;

exports[`projects reducer should handle UPDATE_PROJECT 1`] = `
Array [
  Object {
    "category": "High",
    "description": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
    "editing": false,
    "id": "12252",
    "image": "https://miro.medium.com/max/1000/1*9xsPgsjcFOnwhgPwtsO_HQ.jpeg",
    "title": "Where can I get some?",
  },
  Object {
    "category": "Medium",
    "description": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
    "editing": false,
    "id": "12212352",
    "image": "https://cdn.searchenginejournal.com/wp-content/uploads/2019/08/managing-global-search-projects-760x400.png",
    "title": "Why do we use it?",
  },
  Object {
    "category": "Low",
    "description": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
    "editing": false,
    "id": "23",
    "image": "https://www.oneaccountingsoftware.com/au/wp-content/uploads/2019/08/How-to-Prepare-Annual-Performance-Report.jpg",
    "title": "What is Lorem Ipsum?",
  },
]
`;

exports[`projects reducer should return the initial state 1`] = `
Array [
  Object {
    "category": "High",
    "description": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
    "editing": false,
    "id": "12252",
    "image": "https://miro.medium.com/max/1000/1*9xsPgsjcFOnwhgPwtsO_HQ.jpeg",
    "title": "Where can I get some?",
  },
  Object {
    "category": "Medium",
    "description": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
    "editing": false,
    "id": "12212352",
    "image": "https://cdn.searchenginejournal.com/wp-content/uploads/2019/08/managing-global-search-projects-760x400.png",
    "title": "Why do we use it?",
  },
  Object {
    "category": "Low",
    "description": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
    "editing": false,
    "id": "23",
    "image": "https://www.oneaccountingsoftware.com/au/wp-content/uploads/2019/08/How-to-Prepare-Annual-Performance-Report.jpg",
    "title": "What is Lorem Ipsum?",
  },
]
`;
